staging/src/k8s.io/noderesourcetopology-api/README.md

```diff
@@ -2,9 +2,28 @@
 
 ## Purpose
 This repository contains the CRD based API definition used for enabling NUMA aware Scheduling in Kubernetes. 
-Please refer to [this](https://docs.google.com/document/d/12kj3fK8boNuPNq) document for more details.
+Please refer to [this](https://docs.google.com/document/d/12kj3fK8boNuPNqob6F_pPU9ZTaNEnPGaXEooW1Cilwg/edit?usp=sharing) document for more details.
 
-The repository kubernetes/noderesourcetopology-api is a mirror of https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/noderesourcetopology-api. Please do NOT file issues or submit PRs against the kubernetes/cri-api repository as it is readonly, all development is done in kubernetes/kubernetes.
+The repository kubernetes/noderesourcetopology-api is a mirror of https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/noderesourcetopology-api. Please do NOT file issues or submit PRs against the kubernetes/noderesourcetopology-api repository as it is readonly, all development is done in kubernetes/kubernetes.
+
+## API Definition
+
+API definition can be found in [pkg/apis/topology](pkg/apis/topology).
```

Comment:
These is a very valid question and I am happy to pivot from a CRD based approach to a core API if you think that would be more appropriate. But, as we stand today there are no core in-tree components that rely on this API. This might change down the line as the project matures. I wonder if it is reasonable for us to have a core API that is consumed by out of tree components under kubernetes-sigs org?

For some context, we have been working to validate the Topology-aware scheduling solution on Kubernetes with out of tree components namely [NFD Topology Updater](https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/docs/get-started/introduction.md#nfd-topology-updater) in [kubernetes-sigs/node-feature-discovery](https://github.com/kubernetes-sigs/node-feature-discovery) and [Topology aware scheduler plugin](https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/noderesourcetopology) in [kubernetes-sigs/scheduler-plugins](https://github.com/kubernetes-sigs/scheduler-plugins). These components depend on an API: https://github.com/k8stopologyawareschedwg/noderesourcetopology-api that resides external to kubernetes org . This discussion was reopened because dependency of these two projects (housed under kubernetes-sigs) on an external API was [flagged](https://github.com/kubernetes/community/issues/6308) as not ideal and I wanted to address that.

Donating the [repo](https://github.com/k8stopologyawareschedwg/noderesourcetopology-api) was also proposed as a possible option but I thought proposing a PR and going through an API review would be mean that we can agree on both the API and the approach we should take.

Overall, I am open to suggestions and taking the most suitable path forward.

Reply:
> But, as we stand today there are no core in-tree components that rely on this API.

Then I would not expect it to live in a staging repo.

> These components depend on an API: https://github.com/k8stopologyawareschedwg/noderesourcetopology-api that resides external to kubernetes org . This discussion was reopened because dependency of these two projects (housed under kubernetes-sigs) on an external API was https://github.com/kubernetes/community/issues/6308 as not ideal and I wanted to address that.

Having the API be in a kubernetes or kubernetes-sigs repo makes sense, but if it lives inside the kubernetes/kubernetes repo it's hard for me to see a reason to make it CRD-based.

I haven't followed all the history that got to this point, so maybe I'm missing a key reason to make this API both in-tree and CRD-based, but that seems like the worst of both worlds to me... inflexible and ~slow release cadence, non-standard generation and client/informer location

Reply:
Thanks @liggitt for your input on this.I apologize for the delay in addressing this review comment.

 I understand your point and you make strong case for why we shouldn't be proposing this CRD based API in staging . I am okay with not pursuing this and identifying a more suitable location for this API under the Kubernetes umbrella. I tend to think that moving this API to Kubernetes-sigs makes sense as only out of tree components (NFD and NRT scheduler plugin) depend on this API  and are under kubernetes-sigs. In addition to that,  we have no core in-tree components that currently rely on this API.

I noticed your suggestion about proposing this in Kubernetes and couldn't help but wonder if would be reasonable to proceed in this direction when we don't have in-tree components relying on this core API (at least not yet)? Is there there any precedence of a core API in Kubernetes that has  only out-of-tree API consumers?

Irrespective of the next steps, I think we can pivot this PR to removing the NRT--api from staging. I will update this PR with that.

Reply:
belated lgtm on the PR, sorry for the churn and false starts that led here.

> I noticed your suggestion about proposing this in Kubernetes and couldn't help but wonder if would be reasonable to proceed in this direction when we don't have in-tree components relying on this core API (at least not yet)? Is there there any precedence of a core API in Kubernetes that has only out-of-tree API consumers?

The only ones I can think of are Ingress and NetworkPolicy, though those were added before we had CRDs, and the  the second generation of the Ingress API (GatewayAPI), is being made out of tree with CRDs, so that might not be a great precedent.

Reply:
> belated lgtm on the PR, sorry for the churn and false starts that led here.

No worries. Thanks for your input. I will keep you in the loop and circle back to you (/other API reviewers) when we are converging on the location of the API.

> The only ones I can think of are Ingress and NetworkPolicy, though those were added before we had CRDs, and the the second generation of the Ingress API (GatewayAPI), is being made out of tree with CRDs, so that might not be a great precedent.

Thanks for the information, I see that Ingress API (GatewayAPI) CRD based  API is now under kubernetes-sigs: https://github.com/kubernetes-sigs/gateway-api which aligns with my thought process of proposing this under kubernetes-sigs.